ocrd-olena-binarize

```diff
@@ -1,28 +1,254 @@
 #!/bin/bash
-set -eu
+set -Eu
 # set -x
 
+### binarization with Olena
+# 
+# Finds and downloads all files in the input fileGrp
+# of the workspace. Then for each file, determines its
+# MIME type: 
+# If PAGE-XML, queries the input file for an existing
+# AlternativeImage on the Page level, otherwise uses
+# imageFilename. Passes the image file to the scribo
+# binarization program for the chosen algorithm. The
+# resulting file will reside in OCR-D-IMG-BIN and its
+# ID will be derived from the input file's ID. Finally,
+# adds a reference to the new file by appending some
+# AlternativeImage to the PAGE-XML.
+# If an image, creates a skeleton PAGE-XML for that
+# imageFilename. Passes the image file to scribo
+# likewise. Also adds a reference to the new file
+# by inserting an AlternativeImage to the PAGE-XML.
+# Regardless, adds new image and new PAGE-XML to METS.
+
 which ocrd >/dev/null 2>/dev/null || { echo "ocrd not in \$PATH. Panicking"; exit 1; }
 
 SHAREDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+IMAGE_FILEGRP="OCR-D-IMG-BIN"
+PRESERVE_NAMESPACE=1
+
+function call_scribo {
+    # Binarize as PBM
+    scribo-cli "$1" "$2" "${3}.pbm"
```

Comment:
The problem here is that `scribo-cli`, which is a shell wrapper itself, does not wrap anything but the Sauvola-based algorithms. Wolf and Kim and friends are not available if you fall back to `scribo-cli`. Two options:
1. Modify `scribo-cli` (via `scribo-cli.in` in the Olena repo, not recommended since a merge of a corresponding PR is unlikely to happen)
2. Fall back to the individual processors which are built in `scribo/scribo/src/binarization` if `HAVE_MAGICKXX` is set (via `configure`).

Reply:
I don't think so. The use of `scribo-cli` is what we already had before this PR, and it worked for `wolf` and `kim` as well.

Here is why (from `olena-2.1/scribo/src/scribo-cli`):
```bash
*)
            cmd_bin=`echo $cmd | sed -e 's/-/_/g'`
            if test -f $bdir/$cmd_bin; then
                $bdir/$cmd_bin $@
            else
                echo "Error: Invalid command!"
                usage
            fi
```
So this simply delegates to `local/libexec/scribo/{wolf,kim,niblack,otsu}`.

Reply:
You are absolutely right. It is only the CLI documentation which is restricted to `sauvola`. Sorry!